ENGAGEHF/Bluetooth/Views/MeasurementRecordedView.swift

```diff
@@ -0,0 +1,183 @@
+        }
+    }
+}
+
+
+struct ConfirmMeasurementButton: View {
+    @Binding var viewState: ViewState
+    
+    
+    var body: some View {
+        VStack {
+            AsyncButton(state: $viewState, action: {
+                try await MeasurementManager.manager.saveMeasurement()
+            }) {
+                Text("Save")
+                    .frame(maxWidth: .infinity, maxHeight: 48)
+                    .foregroundStyle(.white)
+                    .background(.accent)
+                    .cornerRadius(8)
+            }
+                .viewStateAlert(state: $viewState)
+            
+            DiscardButton()
+                .disabled(viewState != .idle)
+        }
+    }
+}
+
+
+struct MeasurementRecordedView: View {
+    @Binding var viewState: ViewState
+    
+    var body: some View {
+        VStack {
+            MeasurementHeader(viewState: $viewState)
+            
+            Spacer()
+            
+            MeasurementLayer()
+            
+            Spacer()
+            
+            ConfirmMeasurementButton(viewState: $viewState)
+                .padding()
+            
+            Spacer()
+        }
+    }
+}
+
+
+private struct MeasurementRecordedViewPreviewWrapper: View {
+    @Environment(MeasurementManager.self) private var measurementManager
+    @State private var viewState: ViewState = .idle
+    
+    
+    var body: some View {
+        @Bindable var measurementManager = measurementManager
+        
+        Button("Mock Measurement") {
+            loadMockMeasurement()
+        }
+            .sheet(isPresented: $measurementManager.showSheet) {
+                MeasurementRecordedView(viewState: $viewState)
+            }
+    }
+    
+    
+    private func loadMockMeasurement() {
+        measurementManager.deviceName = "Mock Device"
+        
+        let deviceInformation = DeviceInformationService()
+        deviceInformation.$manufacturerName.inject("Mock")
+        deviceInformation.$modelNumber.inject("42")
+        deviceInformation.$hardwareRevision.inject("42")
+        deviceInformation.$firmwareRevision.inject("42")
+        deviceInformation.$softwareRevision.inject("42")
+        measurementManager.deviceInformation = deviceInformation
+        
+        measurementManager.weightScaleParams = WeightScaleFeature(
+            timeStampEnabled: true,
+            supportMultipleUsers: true,
+            supportBMI: true,
+            weightResolution: .gradeSix,
+            heightResolution: .gradeThree
+        )
+        
+        measurementManager.loadMeasurement(
+            WeightMeasurement(
+                units: .metric,
+                timeStampPresent: true,
+                userIDPresent: true,
+                heightBMIPresent: true,
+                weight: 8400,
+                timeStamp: DateTime(hours: 0, minutes: 0, seconds: 0),
+                bmi: 20,
+                height: 180,
+                userID: 42
+            )
+        )
+    }
```

Comment:
Note on my added code for you @nriedman: This function could also move into an extension of the Measurement service.